service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/S3StorageConnector.java

```diff
@@ -0,0 +1,63 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.S3Object;
+
+public class S3StorageConnector implements StorageConnector {
+
+    private final AmazonS3 s3client;
+    private final String bucketName;
+
+    protected S3StorageConnector(StorageConnectorConfiguration configuration) {
+        this.bucketName = configuration.getContainerName();
+        AWSCredentials credentials = new BasicAWSCredentials(
```

Comment:
For AWS leave out credentials - the AmazonS3 client will read them from the local environments. This lets us just give our Fargate task an IaM role with access to the S3 bucket without managing the keys ourselves.